index.bs

```diff
@@ -2725,6 +2773,249 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
 
 1. Return [=success=] with data null.
 
+</div>
+#### The browsingContext.locateNodes Command ####  {#command-browsingContext-locateNodes}
+
+The <dfn export for=commands>browsingContext.locateNodes</dfn> command returns a
+list of all nodes matching the specified locators.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browsingContext.LocateNodes = (
+        method: "browsingContext.locateNodes",
+        params: browsingContext.LocateNodesParameters
+      )
+
+      browsingContext.LocateNodesParameters = {
+         context: browsingContext.BrowsingContext,
+         locator: browsingContext.Locator,
+         startNodes: [ * script.SharedReference ],
+         ? maxNodes: js-uint,
+      }
+      </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        browsingContext.LocateNodesResult = {
+            context: browsingContext.BrowsingContext,
+            nodeSharedIds: [ * script.SharedId ]
```

Comment:
Still a comment related to the ownership discussion before: would it be possible to return script.NodeListRemoteValue here so that the command can be extended without breaking changes in the future?

Reply:
cc @jgraham @whimboo

Reply:
ah sorry, it does not have a shared id

Reply:
Well, I think what we need here is not a list of shared ids but a list of `NodeRemoteValue`. That will make it way easier later to get ownership support added because a `handle` is already present.

Reply:
Oh, looks like I missed this comment from @jgraham. Nevertheless I can see the benefits by only returning the `sharedId`'s to reduce the amount of data in the payload of the command's response. But it would require the client to construct a `NodeRemoteObject` on its own, also it's not known which special type those nodes have. Also this will be especially hard to use for batched commands where such a returned list cannot easily be used by the next command as actual `NodeRemoteValue`'s.

I've added the `needs-discussion` label so that we can at latest discuss that in the next group meeting if no other consensus has been found before.

Reply:
I think we should return a list of `script.RemoteReference`. Or maybe even `script.RemoteValue` since you can get XPath to return things that aren't actually nodes.

Reply:
Indeed. That's a good point. I agree on that.

Reply:
What about a list of script.NodeRemoteValue then? to indicate that primitive values and non-node objects cannot be returned?

Reply:
@OrKoN @jgraham XPath _can_ return non-nodes from `evaluate`, but we are restricting the return type to be a set of nodes by using `ORDERED_NODE_SNAPSHOT_TYPE` as the result type argument when we call `evaluate`. This still makes sense to me, as the name of the method is, after all, `locateNodes`. I'll remove the restriction around the returned nodes needing to be elements, and I'll return to making the command  return `RemoteNodeValue`s. I'll also return the optional `ownership` argument so that ownership of the resulting nodes can be properly handled by clients.